File: trucking-storm-topology-java/src/main/java/com/orendainx/hortonworks/trucking/storm/schemes/BufferToStringScheme.java
package com.orendainx.hortonworks.trucking.storm.schemes;

import java.nio.ByteBuffer;

import org.apache.storm.tuple.Fields;
import org.apache.storm.tuple.Values;

/**
  * Scheme for parsing speed events.
  *
  * @author Edgar Orendain <edgar@orendainx.com>
  */
public class BufferToStringScheme extends DelimitedScheme {

  private String dataType = "";

  public BufferToStringScheme(String dataType) {
    super("\\|");
  }

  public Values deserialize(ByteBuffer buffer) {
    return new Values(dataType, deserializeAsString(buffer));
  }

  public Fields getOutputFields() {
    return new Fields("dataType","data");
  }
}

File: storm-topology/src/main/scala/com/orendainx/trucking/storm/schemes/BufferToStringScheme.scala
package com.orendainx.trucking.storm.schemes

import java.nio.ByteBuffer

import org.apache.storm.tuple.{Fields, Values}

/**
  * Scheme for parsing speed events.
  *
  * @author Edgar Orendain <edgar@orendainx.com>
  */
class BufferToStringScheme(dataType: String) extends DelimitedScheme("\\|") {

  override def deserialize(buffer: ByteBuffer): Values = new Values(dataType, deserializeAsString(buffer))

  override def getOutputFields: Fields = new Fields("dataType", "data")
}
